platform/darwin/src/NSPredicate+MGLAdditions.h

```diff
@@ -0,0 +1,108 @@
+#import <Foundation/Foundation.h>
+
+#import "NSExpression+MGLAdditions.h"
+#include <mbgl/style/filter.hpp>
+
+@interface NSPredicate (MGLAdditions)
+
+- (mbgl::style::Filter)mgl_filter;
+
+@end
+
+namespace mbgl {
+namespace darwin {
+
+class FilterEvaluator {
```

Comment:
This class is no longer needed. Fold the NSComparisonPredicate operator into `-[NSComparisonPredicate(MGLAdditions) mgl_filter]` and the NSComound Predicate operator into `-[NSCompoundPredicate(MGLAdditions) mgl_filter]`.